### PR TITLE
feat: Add testing to project

### PR DESCRIPTION
**Pull request by roamingthings**
This pull request adds initial tests to the project including the pull request checks.

### CODE REVIEW

File: tests/test_cli.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        '-b', '1.2.3.4',
+        '-p', '1234',
+        '-st', 'streaming-url',
+        '-sn', 'snapshot-url'
+    ])
+    # run_server(bind_address, port, output, stream_url, snapshot_url)StreamingOutput
+    spyglass.server.run_server.assert_called_once_with('1.2.3.4', 1234, ANY, 'streaming-url', 'snapshot-url')
+
+
+def test_start_recording(mocker):
```
Comment (mryel00) [inline tests/test_cli.py:73]: Exactly the same method as `test_start_server`?

File: tests/test_cli.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    mocker.patch('spyglass.server.run_server')
+    from spyglass import cli
+    import spyglass.server
+    cli.main(args=[
+        '-b', '1.2.3.4',
+        '-p', '1234',
+        '-st', 'streaming-url',
+        '-sn', 'snapshot-url'
+    ])
+    # run_server(bind_address, port, output, stream_url, snapshot_url)StreamingOutput
```
Comment (mryel00) [inline tests/test_cli.py:69]: What's up with the `StreamingOutput` at the end of the comment?

File: tests/test_cli.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    cli.main(args=[
+        '-r', '100x200',
+        '-f', '10',
+        '-af', 'manual',
+        '-s', 'fast'
+    ])
+    spyglass.camera.init_camera.assert_called_once_with(100, 200, 10, AF_MODE_ENUM_MANUAL, 0.0, AF_SPEED_ENUM_FAST)
+
+
+def test_start_server(mocker):
```
Comment (mryel00) [inline tests/test_cli.py:57]: Would `test_run_server` make more sense as it is testing `run_server`?

File: .github/workflows/pull-request.yml [hunk 4]
```diff
@@ -24,6 +24,7 @@ jobs:
         pip install flake8 
         pip install build
         if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+        if [ -f requirements.txt ]; then pip install -r requirements-test.txt; fi
```
Comment (mryel00) [inline .github/workflows/pull-request.yml:27]: Wouldn't it make more sense to put both in the same `if` statement if both are using the same condition?

**Review by mryel00 — CHANGES_REQUESTED** [review]

**Comment by roamingthings** [comment]
@mryel00 I've overhauled the CLI tests and workflows

**Review by mryel00 — APPROVED** [review]

**Comment by mryel00** [comment]
Couldn't find anything obviouslyl wrong.

**Pull request merged**